### PR TITLE
Add `change_case()` and `remove_punctuation()` augmentations

### PR DESCRIPTION
**Pull request by kennysong**
Implements `langcheck.augment.change_case()` and `langcheck.augment.remove_punctuation()`.

### CODE REVIEW

File: src/langcheck/augment/en/_remove_punctuation.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    for instance in instances:
+        for _ in range(num_perturbations):
+            perturbed_instance = ''
+            for char in instance:
+                if char not in string.punctuation:
+                    perturbed_instance += char  # No augmentation
+                elif random.random() > aug_char_p:
+                    perturbed_instance += char  # No augmentation
+                else:
+                    pass  # Remove character
```
Comment (Alnusjaponica) [inline src/langcheck/augment/en/_remove_punctuation.py:35]: Same here. You can use `filter` to randomly eliminate punctuations at once.

**Review by Alnusjaponica — COMMENTED** [review]
I made some comments on implementation. Please take a look.

File: src/langcheck/augment/en/_change_case.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        for _ in range(num_perturbations):
+            perturbed_instance = ''
+            for char in instance:
+                if random.random() > aug_char_p:
+                    perturbed_instance += char  # No augmentation
+                else:
+                    if to_case == 'uppercase':
+                        perturbed_instance += char.upper()
+                    else:
+                        perturbed_instance += char.lower()
```
[the remark this answers is not in the document]
Comment (kennysong) [reply src/langcheck/augment/en/_change_case.py:39]: Cool, thanks for looking into this! I think your version still uses 3 loops (2 maps and 1 list comprehension). I [ran a quick benchmark](https://gist.github.com/kennysong/bbbca9defd6c7c63d9b113dae3371778) and it looks like the original version is faster?

In general, I'd prefer to stick with the simpler version until we find a reason to optimize it. I also think our metrics would be the bottleneck for large datasets, not necessarily the data augmentation part.

[the remark this answers is not in the document]
Comment (Alnusjaponica) [reply src/langcheck/augment/en/_change_case.py:39]: > your version still uses 3 loops

Yes, the computational complexity is the same, but `map` is a built-in function and should be faster. As long as the benchmark is not improved, then it's fine. Sorry for taking your time.

**Review by Alnusjaponica — APPROVED** [review]
Thanks for the investigation. LGTM.

File: src/langcheck/augment/en/_change_case.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kennysong) [reply src/langcheck/augment/en/_change_case.py:39]: I guess `map()` might be faster if we don't convert the result into a list? Not really sure though. Let's go with the current approach for now.

**Comment by kennysong** [comment]
Thanks for the review Shinichi!

**Pull request merged**